src/Mvc/Mvc.Razor/src/ApplicationParts/ConsolidatedAssemblyApplicationPartFactory.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    /// <summary>
+    /// Configures an <see cref="ApplicationPart" /> that contains controllers, as well as Razor views and Pages.
+    /// <para>
+    /// Combines the results of <see cref="DefaultApplicationPartFactory.GetApplicationParts(Assembly)"/> and
+    /// <see cref="CompiledRazorAssemblyApplicationPartFactory.GetApplicationParts(Assembly)"/>. This part factory
+    /// may be used if Razor views or Razor Pages are compiled in to with other types including controllers.
+    /// </para>
+    /// </summary>
+    public sealed class ConsolidatedAssemblyApplicationPartFactory : ApplicationPartFactory
```

Comment:
Spicy thought, instead of doing this, we could code-generate this class as part of the compiler work and this way we don't need another public type here.

Reply:
I'm confused as to how this get wires up right now.

Reply:
> Spicy thought, instead of doing this, we could code-generate this class as part of the compiler work

Hmmm....what's the benefit of doing this?

> I'm confused as to how this get wires up right now.

The `ConsolidatedAssemblyApplicationPartFactory` isn't used anywhere at the moment. Eventually, we'll be including the compiled views in the application assembly, which will be stamped stamped with a `ConsolidatedAssembly` attribute, and then using this API to look for `RazorCompiledItem`s that are baked into the app assembly, in addition to it's own types.

@pranavkm Correct me if I got any details wrong.

Reply:
@captainsafia is spot on.

My suggestion is we don't need a public type, we can just codegen one of these puppies per assembly

Reply:
👍🏽 on @captainsafia's analysis. 

@javiercn I'm not sure what code-gening this gets us. Is it to avoid adding a public type?

Reply:
Yep

Reply:
Also because that way, changes in the mechanics of how it works are isolated to the compiler instead of being a concern shared by the runtime

Reply:
I signed-off, but just want you to think about it.

Reply:
> how it works are isolated to the compile

The downside is that the compiler would need to be aware of the app's tfm if we have to change the implementation. We're currently not planning to ship the source generator thru a runtime pack. Once we decide to use it, all of the code it generates is driven by RazorLanguageVersion, not by the app's tfm.

Reply:
let me bring this up during API review. I'm kinda partial to just having the type in the framework, but I'll see how other folks feel about this.